src/clj_kondo/impl/analysis/java.clj

```diff
@@ -45,28 +75,39 @@
               (recur))
             class-name))))))
 
-(defn java-class-def-analysis? [ctx]
-  (-> ctx :config ))
-
 (defn analyze-class-defs? [ctx]
   (:analyze-java-class-defs? ctx))
 
-(defn reg-class-def! [ctx {:keys [jar entry file]}]
+(defn reg-class-def! [ctx {:keys [^JarFile jar ^JarFile$JarFileEntry entry filename ^File file]}]
   (when (analyze-class-defs? ctx)
-    (when-let [class-name (if entry
-                            (entry->class-name entry)
-                            (when file
-                              (cond
-                                (str/ends-with? file ".class")
-                                (class->class-name file)
-                                (str/ends-with? file ".java")
-                                (source->class-name file))))]
-      (swap! (:analysis ctx)
-             update :java-class-definitions conj
-             {:class class-name
-              :uri (->uri jar entry file)
-              :filename (or file
-                            (str jar ":" entry))}))))
+    (let [class-is (or (and jar entry (.getInputStream jar entry))
```

Comment:
I think we need to wrap this in a `with-open` to else we will get too many files open exceptions